packages/vue-vuetify/src/complex/MixedRenderer.vue

```diff
@@ -0,0 +1,491 @@
+      return Array.from(enumTypes).filter((type) => type !== null) as string[];
+    }
+  }
+
+  // return any
+  return ['array', 'boolean', 'integer', 'null', 'number', 'object', 'string'];
+}
+
+const createMixedRenderInfos = (
+  schema: JsonSchema,
+  rootSchema: JsonSchema,
+  control: ControlElement,
+  path: string,
+  uischemas: JsonFormsUISchemaRegistryEntry[],
+): SchemaRenderInfo[] => {
+  let schemas: JsonSchema[] = [];
+
+  if (typeof schema.type === 'string') {
+    schemas.push(schema);
+  } else {
+    const types = getSchemaTypesAsArray(schema);
+
+    types.forEach((type) => {
+      schemas.push({
+        ...schema,
+        type,
+        default:
+          schema.default !== undefined &&
+          type === getJsonDataType(schema.default)
+            ? schema.default
+            : undefined,
+      });
+    });
+  }
+
+  return schemas.map((schema) => {
+    if (schema.type === 'object') {
+      schema.additionalProperties =
+        schema.additionalProperties !== false
+          ? (schema.additionalProperties ?? true)
+          : false;
+    } else if (schema.type === 'array') {
+      schema.items = schema.items ?? {};
+      if ((schema.items as any) === true) {
+        schema.items = {
+          type: [
+            'array',
+            'boolean',
+            'integer',
+            'null',
+            'number',
+            'object',
+            'string',
+          ],
+        };
+      } else if (
+        typeof (schema.items as JsonSchema7).type !== 'string' &&
+        !Array.isArray((schema.items as JsonSchema7).type)
+      ) {
+        (schema.items as JsonSchema7).type = [
+          'array',
+          'boolean',
+          'integer',
+          'null',
+          'number',
+          'object',
+          'string',
+        ];
+      }
+    }
+
+    // help determining the correct renders by removing keywords not appropriate for the type
+    cleanSchema(schema);
+
+    const detailsForSchema = control.options
+      ? control.options[schema.type + '-detail']
+      : undefined;
+
+    const schemaControl = detailsForSchema
+      ? {
+          ...control,
+          options: { ...control.options, detail: detailsForSchema },
+        }
+      : {
+          ...control,
+          scope: control.scope,
+        };
+
+    const uischema = findUISchema(
+      uischemas,
+      schema,
+      control.scope,
+      path,
+      () => createControlElement(control.scope ?? '#'),
+      schemaControl,
+      rootSchema,
+    );
+
+    if (uischema.type === 'Control' && schema.type === 'object') {
+      // override to specify exact type since the isObjectControl will check for empty scope
+      // which in this case the scope is not empty
+      uischema.type = 'Object';
+    }
+
+    return {
+      schema,
+      uischema,
+      label: `${schema.type}`,
+    };
+  });
+};
+
+export function getJsonDataType(value: any): string | null {
```

Comment:
I think we already have a util function like this somewhere in the code base. Likely in the `schema` generator.

Reply:
there is similar function in the core for that but return a schema while here I want the actual type not the schema and the function there is a method - maybe we could exposed that function into the jsonforms core and then reuse that for both cases perhaps